integration/spark-datasource/src/main/spark2.1andspark2.2/org/apache/spark/sql/CarbonVectorProxy.java

```diff
@@ -311,6 +311,7 @@ public void setDictionary(CarbonDictionary dictionary) {
         vector.setDictionary(dictionaryWrapper);
         this.dictionary = dictionaryWrapper;
       } else {
+        this.dictionary = null;
```

Comment:
This change lies in spark2.1andspark2.2, so spark2.3 does not have this problem？

Reply:
In 2.3 proper interface are exposed from spark and same has been implemented, so its not required